src/plugins/intel_gna/orientation_helper.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "orientation_helper.hpp"
+
+#include <legacy/ie_layers.h>
+
+#include <gna_graph_tools.hpp>
+
+namespace GNAPluginNS {
+namespace helpers {
+
+void updateModelInputOrientationWithoutConvolution(const InferenceEngine::CNNLayer& inputLayer,
+                                                   const GNAPluginNS::backend::DnnComponents& components,
+                                                   GNAPluginNS::GnaInputs& inputs) {
+    // does not make sense to go further is there is no input to set
+    auto input = inputs.find(inputLayer.name);
+
+    if (input == inputs.end()) {
+        return;
+    }
+
+    auto doesntHaveGnaMapping = [=](InferenceEngine::CNNLayerPtr l) {
+        auto dnnLayer = components.findComponent(l);
+        return dnnLayer == nullptr;
+    };
+
+    auto nextLayers = InferenceEngine::CNNNetGetAllNextLayersSkipCertain(&inputLayer, -1, doesntHaveGnaMapping);
+    if (nextLayers.empty()) {
+        input->orientation = kDnnInterleavedOrientation;
+        return;
+    }
+
+    std::vector<intel_dnn_orientation_t> suggestedOrientations;
+
+    auto dims = input->dims;
+    auto raws = dims[0];
+    auto rest = InferenceEngine::details::product(std::next(std::begin(dims)), std::end(dims));
+    if (rest == 0) {
+        rest = 1;
+    }
+
+    for (auto& nextLayer : nextLayers) {
+        auto dnnLayer = components.findComponent(nextLayer);
+
+        if (!dnnLayer) {
+            THROW_GNA_LAYER_EXCEPTION(nextLayer) << " gna mapped layer search connection failed";
+        }
+
+        if (dnnLayer->operation != kDnnInterleaveOp && dnnLayer->operation != kDnnDeinterleaveOp && raws > 1 &&
```

Comment:
`raws` and `rest` are not updated inside loop. Does it mean that we can skip loop if condition is not satisfied?

Reply:
they are dependent to the input dims - there is one input only at the same time- we iterate over the layers only.

Reply:
> they are dependent to the input dims - there is one input only at the same time- we iterate over the layers only.

I see. I mean that if either raws == 1 either rest == 1 we can skip loop at all

Reply:
can we reevaluate this comment?